tmcprototype/CspSubarrayLeafNode/CspSubarrayLeafNode/CspSubarrayLeafNode.py

```diff
@@ -124,6 +131,75 @@ def commandCallback(self, event):
+        considered for delay calculations. The values of receptors and frequency slice
+        ids are also non-standard.
+        The epoch value is the current timestamp value.
+
+        :param argin: int.
+            The argument contains delay model update interval in seconds.
+
+        :return: None.
+        """
+        delay_update_interval = argin
+        # list of frequency slice ids
+        _fsids_list = ["fs1", "fs2", "fs3", "fs4"]
+        # list of bands
+        _bands_list = ["band1", "band2", "band3", "band4", "band5a", "band5b"]
+        # receptor list
+        # TBD: This list should be taken from receptor_id_list attribute of subarray node
+        _receptor_list = ["d0001", "d0002", "d0003", "d0004"]
+
+        while not self._stop_delay_model_event.isSet():
+            if(self.CspSubarrayProxy.obsState == CONST.ENUM_READY
+                    or self.CspSubarrayProxy.obsState == CONST.ENUM_SCANNING):
+
+                self.dev_logging("Calculating delays.", int(tango.LogLevel.LOG_INFO))
+                delay_model_json = {}
+                delay_model = []
+                receptor_delay_model = []
+                delay_model_per_epoch = {}
+                for receptor in _receptor_list:
+                    receptor_delay_object = {}
+                    receptor_delay_object["receptor"] = receptor
+                    receptor_specific_delay_details = []
+                    for fsid in _fsids_list:
+                        fsid_delay_object = {}
+                        fsid_delay_object["fsid"] = fsid
+                        delay_coeff_array = []
+                        for band in _bands_list:
+                            delay_coeff_array.append(random.uniform(0.01, 10))  # random delay
+                        fsid_delay_object["delayCoeff"] = delay_coeff_array
+                        receptor_specific_delay_details.append(fsid_delay_object)
+                    receptor_delay_object["receptorDelayDetails"] = receptor_specific_delay_details
+                    receptor_delay_model.append(receptor_delay_object)
+                delay_model_per_epoch["epoch"] = calendar.timegm(time.gmtime())
+                # delay_model_per_epoch["epoch"] = (datetime.datetime.now() + datetime.timedelta(seconds=5)).timestamp()
+                delay_model_per_epoch["delayDetails"] = receptor_delay_model
+                delay_model.append(delay_model_per_epoch)
+                delay_model_json["delayModel"] = delay_model
+                print("delay_model_json: ", delay_model_json)
+
+                # update the attribute
+                self.delay_model_lock.acquire()
+                self._delay_model = json.dumps(delay_model_json)
+                self.delay_model_lock.release()
+
+                # wait for timer event
+                self._stop_delay_model_event.wait(delay_update_interval)
+            else:
+                self._delay_model = " "
+
+            print("Stop event received. Thread exit.")
```

Comment:
The last 2 statements are inside loop. They should be taken out.

Reply:
Yes. Done.